internal/pkg/githubapi/github.go

```diff
@@ -284,15 +285,81 @@ func handleChangedPREvent(ctx context.Context, mainGithubClientPair GhClientPair
 	return nil
 }
 
+func buildArgoCdDiffComment(diffCommentData DiffCommentData, beConcise bool, partNumber int, totalParts int) (string, error) {
+	buf := new(bytes.Buffer)
+	mb := md.NewMarkdown(buf)
+	const argoSmallLogo = `<img src="https://argo-cd.readthedocs.io/en/stable/assets/favicon.png" width="20"/>`
+	if partNumber != 0 {
+		mb.PlainTextf("Component %d/%d: %s (Split for comment size)\n", partNumber, totalParts, diffCommentData.DiffOfChangedComponents[0].ComponentPath)
+	}
+	if !beConcise {
+		mb.PlainText("Diff of ArgoCD applications:\n")
+	} else {
+		mb.PlainText("Diff of ArgoCD applications(concise view, full diff didn't fit GH comment):\n")
+	}
+	for _, appDiffResult := range diffCommentData.DiffOfChangedComponents {
+		if appDiffResult.DiffError != nil {
+			mb.Cautionf("%s (%s) ", md.Bold("Error getting diff from ArgoCD"), md.Code(appDiffResult.ComponentPath))
+			mb.PlainTextf("Please check the App Conditions of %s %s for more details.", argoSmallLogo, md.Bold(md.Link(appDiffResult.ArgoCdAppName, appDiffResult.ArgoCdAppURL)))
+			if appDiffResult.AppWasTemporarilyCreated {
+				mb.Warning("For investigation we kept the temporary application, please make sure to clean it up later!")
+			}
+			mb.CodeBlocks(md.SyntaxHighlightNone, appDiffResult.DiffError.Error())
+		} else {
+			mb.PlainTextf("%s %s @ %s", argoSmallLogo, md.Bold(md.Link(appDiffResult.ArgoCdAppName, appDiffResult.ArgoCdAppURL)), md.Code(appDiffResult.ComponentPath))
+
+			// If the app was temporarily created, we should inform the user about it, if not we should inform about "unusual" health and sync status
+			if appDiffResult.AppWasTemporarilyCreated {
+				mb.Note("Telefonistka has temporarily created an ArgoCD app object to render manifest previews.  \nPlease be aware:  \n* The app will only appear in the ArgoCD UI for a few seconds.")
+			} else {
+				if appDiffResult.ArgoCdAppHealthStatus != "Healthy" {
+					mb.Cautionf("The ArgoCD app health status is currently %s", appDiffResult.ArgoCdAppHealthStatus)
+				}
+				if appDiffResult.ArgoCdAppSyncStatus != "Synced" {
+					mb.Warningf("The ArgoCD app sync status is currently %s", appDiffResult.ArgoCdAppSyncStatus)
+				}
+				if !appDiffResult.ArgoCdAppAutoSyncEnabled {
+					mb.Note("This ArgoCD app is doesn't have `auto-sync` enabled, merging this PR will **not** apply changes to cluster without additional actions.")
```

Comment:
Probably not with this PR but it would be if we can find a way to also inject documentation links, dynamically with an env configuration.
For example "how to enable auto-sync?"